department-of-veterans-affairs/caseflow
# Fix bug where editing the transcription details would cause virtual hearings notifications to display

Related: https://github.com/department-of-veterans-affairs/caseflow/pull/14747

### Description

- Fix an issue where a payload value of `false` would prevent the user alerts from displaying
- Refactor `DetailsContainer ` into a functional component, and remove it's connection the Redux store, as well as an extra `loading` state key.
- Allow `Details` to fully manage the `HearingsFormContext` rather than deferring part of it to `DetailsContainer` (through the `setHearing` prop)
- Fix an issue where `formatHearing` wasn't applied to the `initialHearing` in the reset action

### Testing Plan
1. Login as a Hearing Coordinator
2. Navigate to a virtual hearing
3. Edit the transcription details, and make sure you don't see an extra virtual hearings notification
4. Stay on the page, and repeat step 3 a few times with different fields, and ensure the behavior is consistent
